packages/sanity/src/desk/structureBuilder/util/getStructureBuilderId.ts

```diff
@@ -0,0 +1,16 @@
+import {camelCase} from 'lodash'
+import getSlug from 'speakingurl'
+import {disallowedPattern} from './validateId'
+
+export function getStructureBuilderId(title: string, id?: string): string {
+  if (id) {
+    return id
+  }
+
+  const [disallowedChar] = title.match(disallowedPattern) || []
+  if (disallowedChar) {
+    return camelCase(getSlug(title))
+  }
+
+  return camelCase(title)
```

Comment:
Two things here:

1. The matching must be tested against the camelcased version, since the title is likely to include spaces and similar. Otherwise I think this would match too often. Not the end of the world, but it would make some backwards incompatible changes to existing IDs that may be bookmarked already.
2. I think it is probably easier to use `regex.test()` here, since we're not using the matched character

```suggestion
  const camelCased = camelCase(title)
  return disallowedPattern.test(camelCased) ? camelCase(getSlug(title)) : camelCased
```